webserver/src/main/java/io/kestra/webserver/services/BasicAuthService.java

```
@@ -47,22 +47,21 @@ public class BasicAuthService {
 
     @PostConstruct
     private void init() {
-        if (Boolean.TRUE.equals(this.basicAuthConfiguration.getEnabled())) {
-            this.save(this.basicAuthConfiguration);
-        } else if (Boolean.FALSE.equals(this.basicAuthConfiguration.getEnabled())) {
-            this.unsecure();
+        if (basicAuthConfiguration == null){
+            settingRepository.save(Setting.builder()
+                .key(BASIC_AUTH_SETTINGS_KEY)
+                .value(new BasicAuthConfiguration())
+                .build());
+        } else if (basicAuthConfiguration.getUsername() == null || basicAuthConfiguration.getPassword() == null){
+            settingRepository.save(Setting.builder()
+                .key(BASIC_AUTH_SETTINGS_KEY)
+                .value(basicAuthConfiguration)
+                .build());
```

Why you don't let the validation throw?